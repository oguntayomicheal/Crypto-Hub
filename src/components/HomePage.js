import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import CoinList from './CoinList';
import fetchProducts from '../redux/Data/Data';
import Navbar from './Navbar';
import style from './css/HomePage.module.css';

const HomePage = () => {
  const dispatch = useDispatch();

  const allCoins = useSelector((state) => state.data);
  const [search, setSearch] = useState('');

  useEffect(() => {
    dispatch(fetchProducts());
  }, [dispatch]);

  return (
    <>
      <img src="./cryptoHub.jpg" alt="cryptoHub" className={style.bannerImg} />
      <Navbar />
      <div className={style.inputCtn}>
        <input
          type="text"
          value={search}
          onChange={(e) => setSearch(e.target.value)}
          placeholder="Search or filter coins by name or symbol"
          className={style.input}
        />
      </div>

      <div className={style.allCoins}>
        {
          allCoins.filter((searchCoin) => searchCoin.name.toLowerCase()
            .includes(search.toLowerCase())
            || searchCoin.symbol.toLowerCase().includes(search.toLowerCase()))
            .map((coins) => (
              <div key={coins.id} className={style.eachCoin}>
                <CoinList
                  key={coins.id}
                  img={coins.image}
                  id={coins.id.toUpperCase()}
                  name={coins.name}
                  symbol={coins.symbol.toUpperCase()}
                  atl={coins.atl}
                  price={coins.current_price}
                  details={coins}
                />
              </div>
            ))
        }
      </div>

    </>
  );
};

export default HomePage;
