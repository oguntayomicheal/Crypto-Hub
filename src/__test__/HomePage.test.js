import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import HomePage from '../components/HomePage';

describe('testing Home page', () => {
  it('should render the Home page correctly', () => {
    const Home = render(
      <Router>
        <Provider store={store}>
          <HomePage />
        </Provider>
      </Router>,
    );
    expect(Home).toMatchSnapshot();
  });
});
