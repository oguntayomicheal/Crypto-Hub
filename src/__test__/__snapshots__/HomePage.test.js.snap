// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Home page should render the Home page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="cryptoHub"
        class="bannerImg"
        src="./cryptoHub.jpg"
      />
      <nav
        class="navbar"
      >
        <a
          class="Nav"
          href="/"
        >
          CryptoHub
        </a>
        <div
          class="navIcon"
        >
          <i
            class="fa fa-gear"
          />
          <i
            class="fa fa-microphone"
          />
        </div>
      </nav>
      <div
        class="inputCtn"
      >
        <input
          class="input"
          placeholder="Search or filter coins by name or symbol"
          type="text"
          value=""
        />
      </div>
      <div
        class="allCoins"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="cryptoHub"
      class="bannerImg"
      src="./cryptoHub.jpg"
    />
    <nav
      class="navbar"
    >
      <a
        class="Nav"
        href="/"
      >
        CryptoHub
      </a>
      <div
        class="navIcon"
      >
        <i
          class="fa fa-gear"
        />
        <i
          class="fa fa-microphone"
        />
      </div>
    </nav>
    <div
      class="inputCtn"
    >
      <input
        class="input"
        placeholder="Search or filter coins by name or symbol"
        type="text"
        value=""
      />
    </div>
    <div
      class="allCoins"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
